src/dotnet-restore/Program.cs

```diff
@@ -124,7 +124,7 @@ private static void RestoreTools(Project project, RestoreTask restoreTask, bool
 
         private static void RestoreTool(LibraryRange tooldep, RestoreTask restoreTask, bool quiet)
         {
-            var tempRoot = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+            var tempRoot = Path.Combine(restoreTask.ProjectDirectory, Guid.NewGuid().ToString());
```

Comment:
At line [130](https://github.com/dotnet/cli/pull/898/files#diff-67e89a17951b6cc7bb7ddaee1f4620a9R130) it is indeed get put in the bin folder.